resolver.go

```diff
@@ -184,7 +189,14 @@ func createRespMsg(query *dns.Msg) *dns.Msg {
 }
 
 func (r *resolver) handleIPQuery(name string, query *dns.Msg, ipType int) (*dns.Msg, error) {
-	addr, ipv6Miss := r.sb.ResolveName(name, ipType)
+	var addr []net.IP
+	var ipv6Miss bool
+	if r.sb != nil {
+		addr, ipv6Miss = r.sb.ResolveName(name, ipType)
+	} else if r.network != nil {
```

Comment:
We should avoid this repeated if/else checks by keeping the lookup functions defined on the sandbox. Please see the earlier comment.